src/rpdk/core/data_loaders.py

```diff
@@ -106,6 +113,16 @@ def load_resource_spec(resource_spec_file):  # noqa: C901
         LOG.debug("Resource spec validation failed", exc_info=True)
         raise SpecValidationError(str(e)) from e
 
+    in_readonly = _is_in(resource_spec, "readOnlyProperties")
+    in_createonly = _is_in(resource_spec, "createOnlyProperties")
+
+    primary_id = resource_spec["primaryIdentifier"]
+    if not in_readonly(primary_id) and not in_createonly(primary_id):
```

Comment:
can simplify a bit to make condition `primaryIdentifier - (createOnlyProperties | readOnlyProperties)`  where they are all sets

Reply:
used closure to reuse the subset operation but we might use a disjoint of `createOnly | readOnly`

Reply:
[not sure this is handling composite primaryIdentifiers correctly](https://sage.amazon.com/questions/1025861)